tests/phpunit/api/v3/FilterCaseByStatus.php

```diff
@@ -0,0 +1,73 @@
+ | Copyright CiviCRM LLC (c) 2004-2015                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * Include class definitions
+ */
+require_once 'CiviTest/CiviCaseTestCase.php';
+
+/**
+ * Test whether the case api filters results by status id when searching using contact id.
+ *
+ * @package CiviCRM_APIv3
+ */
+class api_v3_FilterCaseByStatus extends CiviCaseTestCase {
+  public function testCaseGetByStatus() {
+    // Create 2 cases with different status ids.
+    $case1 = $this->callAPISuccess('Case', 'create', array(
+        'contact_id' => 17,
+        'subject' => "Test case 1",
+        'case_type_id' => $this->caseTypeId,
+        'status_id' => "Open",
+        'sequential' => 1,
+      ));
+
+    $case2 = $this->callAPISuccess('Case', 'create', array(
+        'contact_id' => 17,
+        'subject' => "Test case 2",
+        'case_type_id' => $this->caseTypeId,
+        'status_id' => "Urgent",
+        'sequential' => 1,
+      ));
+
+    // Get cases by contact id. Should return both the cases.
+    $result1 = $this->callAPISuccess('Case', 'get', array(
+        'contact_id' => 17,
+        'sequential' => 1,
+      ));
+
+    // Get cases by contact id and status id. Should return only case 1.
+    $result2 = $this->callAPISuccess('Case', 'get', array(
+        'sequential' => 1,
+        'contact_id' => 17,
+        'status_id' => "Open",
+      ));
+
+    // This should clearly fail if the api was working properly. However, this test passes.
+    $this->assertAPIArrayComparison($result1['values'], $result2['values']);
```

Comment:
If this test fails to verify these two results are equal, i.e. only case1 is retrieved in result2. We will be done.